lib/builderator/interface.rb

```diff
@@ -10,20 +10,36 @@ module Builderator
   ##
   # Base class for integration interfaces
   ##
-  class Interface < Config::Attributes
+  class Interface
     class << self
+      def command(arg = nil)
+        @command = arg unless arg.nil?
+        @command
+      end
+
+      def from_gem(arg = nil)
+        @from_gem = arg unless arg.nil?
+        @from_gem || @command
+      end
+
       def template(arg = nil)
         @template = arg unless arg.nil?
         @template
       end
     end
 
-    def initialize(*_)
-      ## Ensure that Config is compiled before reading from it
-      Config.recompile
+    ## Is vagrant in this bundle?
+    def bundled?
+      Gem.loaded_specs.key?(self.class.from_gem)
+    end
+
+    def which
+      return self.class.command if bundled?
 
-      super
+      ## Not in the bundle. Use system path
+      `which #{self.class.command}`.chomp.tap { |path| fail "Unable to locate a #{self.class.command} executable" if path.empty? }
     end
+    alias_method :command, :which
```

Comment:
You can you `mkmf` for this.

http://ruby-doc.org/stdlib-2.0.0/libdoc/mkmf/rdoc/MakeMakefile.html#method-i-find_executable